api/nfd/v1alpha1/types.go

```diff
@@ -139,6 +141,36 @@ type NodeFeatureRuleSpec struct {
 	Rules []Rule `json:"rules"`
 }
 
+// NodeFeatureRuleStatus represents the status of a NodeFeatureRule
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced,shortName=nfs
+// +kubebuilder:subresource:status
```

Comment:
These should be organized under the parent object NodeFeatureRule. Since they already exist, you could just drop them. You can find the relevant section here: https://github.com/kubernetes-sigs/node-feature-discovery/blob/955095c7eb2e7afe2276cd0b6a2dab44042c5ded/api/nfd/v1alpha1/types.go#L123. .